toolkits/google_sheets/arcade_google_sheets/utils.py

```diff
@@ -546,3 +547,212 @@ def parse_write_to_cell_response(response: dict) -> dict:
+    )
+    spreadsheet = Spreadsheet.model_validate(metadata_response)
+
+    target_sheet = find_sheet_by_identifier(spreadsheet.sheets, sheet_identifier)
+    if not target_sheet:
+        raise ToolExecutionError(
+            message=f"Sheet with identifier '{sheet_identifier}' not found",
+            developer_message=(
+                "Sheet(s) in the spreadsheet: "
+                + ", ".join([sheet.model_dump_json() for sheet in spreadsheet.sheets])
+            ),
+        )
+
+    a1_ranges = []
+    sheet_name = target_sheet.properties.title
+    grid_props = target_sheet.properties.gridProperties
+    if grid_props:
+        sheet_row_count = grid_props.rowCount
+        sheet_col_count = grid_props.columnCount
+
+        curr_range = calculate_a1_sheet_range(
+            sheet_name,
+            sheet_row_count,
+            sheet_col_count,
+            start_row,
+            start_col,
+            max_rows,
+            max_cols,
+        )
+        if curr_range:
+            a1_ranges.append(curr_range)
+
+    # Next, get the data for the ranges
+    if a1_ranges:
+        response = (
+            service.spreadsheets()
+            .get(
+                spreadsheetId=spreadsheet_id,
+                includeGridData=True,
+                ranges=a1_ranges,
+                fields="spreadsheetId,spreadsheetUrl,properties/title,sheets/properties,sheets/data/rowData/values/userEnteredValue,sheets/data/rowData/values/formattedValue,sheets/data/rowData/values/effectiveValue",
+            )
+            .execute()
+        )
+    else:
+        response = metadata_response
+
+    return parse_get_spreadsheet_response(response)
+
+
+def process_get_spreadsheet_params(
+    start_row: int,
+    start_col: str,
+    max_rows: int,
+    max_cols: int,
+) -> tuple[int, str, int, int]:
+    """Process and validate the input parameters for the get_spreadsheet tool.
+
+    Args:
+        start_row (int): Processed to be within the range [1, 1000]
+        start_col (str): Processed to be alphabetic column representation. e.g., A, Z, QED
+        max_rows (int): Processed to be within the range [1, 1000]
+        max_cols (int): Processed to be within the range [1, 26]
+
+    Returns:
+        tuple[int, str, int, int]: The processed parameters.
+
+    Raises:
+        ToolExecutionError:
+            If the start_col is not one of alphabetic or numeric
+    """
+    processed_start_row = max(1, start_row)
+    processed_max_rows = max(1, min(max_rows, 1000))
+    processed_max_cols = max(1, min(max_cols, 26))
+    if not start_col.isalpha():
+        if not start_col.isdigit():
+            raise ToolExecutionError("Input 'start_col' must be alphabetic")
+        processed_start_col = index_to_col(int(start_col) - 1)
+    else:
+        processed_start_col = start_col.upper()
+
+    return processed_start_row, processed_start_col, processed_max_rows, processed_max_cols
+
+
+def enforce_data_size_limit(data: dict) -> None:
```

Comment:
Since this only raises an exception if the data is too large, I'd rename it to something like `raise_exception_for_large_payload`.

The name suggested me it would going to trim the data to "enforce" it to comply with a certain size threshold - which would probably not be a good idea :)